src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -1893,9 +1902,16 @@ public void CheckLibraryImportsUpgrade ([Values(false, true)] bool useShortFileN
 					Assert.IsTrue (Directory.Exists (Path.Combine (Root, path, proj.ProjectName, proj.IntermediateOutputPath, "__library_projects__")),
 						"The __library_projects__ directory should exist.");
 					proj.RemoveProperty ("_AndroidLibrayProjectIntermediatePath");
+					//HACK: forces project to re-save on Windows, is there a *better* way?
+					proj.Sources.First ().Timestamp = DateTimeOffset.UtcNow;
```

Comment:
@dellis1972 let me know if there is a better way to force this to re-save.

It seemed that the project file's "timestamp" is based on the newest containing file, so property changes have no effect.

Reply:
@jonathanpeppers the builder.Build should take a `saveProject` property which will force a project save.  But it looks like we still only use the latest timestamp.. 
We should probably introduce a project timestamp at some point to remove this hack.